File: json/crud_simples/crud.js
const fs = require("fs");
const promptSync = require("prompt-sync")();

//Cria o menu de opções::

function menu(){
    console.log("\nMenu de Contatos: ")
    console.log("1 - Cadastrar Contato")
    console.log("2 - Listar Contatos")
    console.log("3 - atualizar Contato")
    console.log("4 - Excluir Contato")
    console.log("5 - Sair")
}

function main(){
    let opcao;
    do {
        menu();
        opcao = promptSync("Escolha uma opção: ");
        switch(opcao){
            case "1":
                adicionar(); // chama a função para adicionar contato
                break;
            case "2":
                listar(); // chama a função para listar contatos
                break;
            case "3":
                atualizar(); // chama a função para atualizar contato
                break;
            case "4":
                excluir(); // chama a função para excluir contato
                break;
            case "5":
                console.log("Saindo...");
                break;
            default:
                console.log("Opção inválida!");
        }
    } while(opcao !== "5");
}
// função para ler os dados do arquivo JSON

function LerDados(){
    const dados = fs.readFileSync("contatos.json", "utf-8"); 
    return JSON.parse(dados || "[]"); // retorna um array vazio se não houver dados no JSON
}

// função para adicionar um novo contato
function adicionar(){
    nome = promptSync("Digite o nome do contato: ");
    telefone = promptSync("Digite o telefone do contato: ");

    const contatos = LerDados(); // lê os contatos existentes
    contatos.push({nome, telefone});
    salvarDados(contatos); //salva os contatos atualizados no arquivo JSON
    console.log("Contato adicionado com sucesso!");
}

// Função para listar os contatos
function listar(){
    const contatos = LerDados(); // lê os contatos existentes
    console.log("Lista de Contatos:");
    contatos.forEach((contato, index) => {
        console.log(`${index + 1}. ${contato.nome} - ${contato.telefone}`);
    });
}
// Função para atualizar um contato
function atualizar(){
    const indexAtualizar = parseInt(promptSync("Digite o número do contato que deseja atualizar: ")) - 1;
    const contatos = LerDados();
    if(indexAtualizar >= 0 && indexAtualizar < contatos.length){
        const novoNome = promptSync("Digite o novo nome do contato: ");
        const novoTelefone = promptSync("Digite o novo telefone do contato: ")

        contatos[indexAtualizar] = {nome: novoNome, telefone: novoTelefone}; // atualiza o contato
        salvarDados(contatos); // salva os contatos atualizados no arquivo JSON
        console.log("Contato atualizado com sucesso!");
    }else{
        console.log("Contato não encontrado!");
    }
}

function excluir(){
    const indexExcluir = parseInt(promptSync("Digite o número do contato que deseja excluir: ")) - 1;
    const contatos = LerDados();
    if (indexExcluir >= 0 && indexExcluir < contatos.length){
        contatos.splice(indexExcluir, 1); // remove o contato do array
        salvarDados(contatos); // salva os contatos atualizados no arquivo JSON
        console.log("Contato excluído com sucesso!");
    }else{
        console.log("Contato não encontrado!");
    }
}

// função para "salvar" os dados no JSON
function salvarDados(contatos){
    fs.writeFileSync("contatos.json", JSON.stringify(contatos, null, 2)); // salva os contatos no arquivo JSON
}


main(); // inicia o programa

File: json/Python/CRUD_Simples/crud.py
import json
import os

CAMINHO = r"C:\Users\2DevSESI\Documents\LuanBasani\backend\Jorge\json\Python\dados.json"

def carregarDados():
    if not os.path.exists(CAMINHO):
        return []
    with open(CAMINHO, "r", encoding="utf-8") as arquivo:
        return json.load(arquivo)

def salvar_dados(dados):
    with open(CAMINHO, "w", encoding="utf-8") as arquivo:
        json.dump(dados, arquivo, indent=4, ensure_ascii=False)
    
def criar_pessoa(nome,idade):
    dados = carregarDados()

    #gera um id simples
    novo_id = 1
    if dados:
        novo_id = dados[-1]["id"] + 1
    pessoa = {
        "id": novo_id,
        "nome": nome,
        "idade": idade
    }

    dados.append(pessoa)
    salvar_dados(dados)
    print("Pessoa cadastrada com sucesso!")

def listar_pessoas():
    dados = carregarDados()

    if not dados:
        print("Nenhuma pessoa cadastrada.")
        return
    
    for pessoa in dados:
        print(f"ID: {pessoa['id']} | Nome: {pessoa['nome']} | Idade: {pessoa['idade']}  ")


def atualizar_pessoa(id, novo_nome, nova_Idade):
    dados = carregarDados()

    for pessoa in dados:
        if pessoa["id"] == id:
            pessoa["nome"] = novo_nome
            pessoa["idade"] = nova_Idade
            salvar_dados(dados)
            print("Pessoa atualizada com sucesso!")
            return
    print("ID não encontrada.")

def deletar_pessoa(id):
    dados = carregarDados()
    #cria nova lista sem o id informado
    dados = [pessoa for pessoa in dados if pessoa ["id"] != id]
    salvar_dados(dados)
    print("Pessoa deletada com sucesso!")

def main():
    while True:
        print("\nEscolha uma opção:")
        print("\n1 - Cadastrar Pessoa")
        print("2 - Listar Pessoas")
        print("3 - Atualizar Pessoa")
        print("4 - Deletar Pessoa")
        print("0 - Sair")

        opcao = input("Escolha: ")
        if opcao == "1":
            nome = input("Nome: ")
            idade = int(input("Idade: "))
            criar_pessoa(nome,idade)

        elif opcao == "2":
            listar_pessoas()

        elif opcao == "3":
            id = int(input("ID da pessoa a ser atualizada: "))
            novo_nome = input("Novo nome: ")
            nova_idade = int(input("Nova idade: "))
            atualizar_pessoa(id, novo_nome, nova_idade)

        elif opcao == "4":
            id = int(input("ID da pessoa a ser deletada: "))
            deletar_pessoa(id)

        elif opcao == "0":
            print("Saindo do programa...")
            break

        else:
            print("Opção inválida. Tente novamente.")

if __name__ == "__main__":
    main()